libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -940,15 +971,39 @@ void AC_PosControl::update_z_controller()
 
     // Velocity Controller
 
+    // Constrain to configured hard limits
```

Comment:
The vertical speed limit is done in the _p_pos_z object. So I don't think this is where we would limit this if we decide to do this.

Reply:
I disagree. The position controller can only ask for whatever the configured target speed is. If there is no velocity feed-forward that is fine, the vehicle will travel at a maximum of configured target speed. The problem comes when the combination of demand from the position controller and velocity feed forward combine to result in speed that is outside the expected range.

You have highlighted a error in my assumption that the overspend is un-constrained, due to the position controller limit the maximum overspeed is double the target speed, which is exactly what I saw in my testing. However the acceleration "overspeed" is unconstrained (until you reach high or low throttle saturation).

Reply:
The problem is the misunderstanding that the "expected range" is the maximum navigation speed. The expected range is double your maximum navigation speed, 1 for command and 1 for correction. You always need to be able to move faster than your navigation maximum or you can never correct a position error.

Reply:
Yes, I agree that the unconstrained acceleration is the problem here. The assumption was we wanted to access everything we could, limited only by thrust, to stay airborne.

Reply:
I think double the max navigation speed is too high, if we were going to hard code it I would say something like 1.25 would be more sensible. The advantage of adding the parameters is that they default to 0 for off so there is no change, but as you say it can mean that the limits could be set such that the vehicle can never keep up, we could get around this with arming checks. 

This should not change the maximum thrust we can use just instead of that coming in from the higher level controllers extreme acceleration demand we instead rely on the acceleration controllers I term to boost throttle to meet the limited demand.

Reply:
> This should not change the maximum thrust we can use just instead of that coming in from the higher level controllers extreme acceleration demand we instead rely on the acceleration controllers I term to boost throttle to meet the limited demand.

The concern here is that I term can be much slower and cause oscillation or overshoot (not good when going down).